app/pages/project/index.spec.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+import apiClient from 'panoptes-client/lib/api-client';
+import { ProjectPageController } from './';
+
+const params = {
+  owner: 'test',
+  name: 'test'
+};
+
+const location = {
+  query: {}
+};
+
+const context = {
+  initialLoadComplete: true,
+  router: {}
+};
+
+const resources = {
+  pages: [],
+  project_roles: [
+    {
+      roles: ['owner'],
+      links: {
+        owner: {
+          id: '1'
+        }
+      }
+    },
+    {
+      roles: ['collaborator'],
+      links: {
+        owner: {
+          id: '2'
+        }
+      }
+    },
+    {
+      roles: ['tester'],
+      links: {
+        owner: {
+          id: '3'
+        }
+      }
+    }
+  ]
+};
+
+const project = apiClient.type('projects').create({
+  id: '1',
+  display_name: 'A test project',
+  experimental_tools: [],
+  get(type) {
+    return Promise.resolve(resources[type]);
+  },
+  links: {
+    active_workflows: [1, 2, 3, 4, 5],
+    avatar: { id: '1' },
+    background: { id: '1' },
+    owner: { id: '1' }
+  }
+});
+
+const background = apiClient.type('backgrounds').create({
+  id: project.links.background.id
+});
+
+const owner = apiClient.type('users').create({
+  id: project.links.owner.id
+});
+
+const projectAvatar = apiClient.type('avatars').create({
+  id: project.links.avatar.id
+});
+
+const preferences = apiClient.type('project_preferences').create({
+  preferences: {},
+  links: {
+    project: project.id
+  },
+  save() {
+    Promise.resolve(preferences);
```

Comment:
Can you reference a variable inside its own declaration? Also, the promise should be returned if this is mocking the API client.

Reply:
This was added to stop `preference.save()` throwing warnings somewhere in the component. I don't think the return value is actually used in the code, so it might be safer to just return `null` here.

Reply:
JavaScript is fine with circular references inside a declaration, but it probably is a terrible thing to do.